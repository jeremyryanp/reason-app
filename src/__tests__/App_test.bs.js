// Generated by BUCKLESCRIPT VERSION 4.0.5, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("@glennsl/bs-jest/src/jest.js");
var Enzyme = require("enzyme");
var ReasonReact = require("reason-react/src/ReasonReact.js");
var App$ReasonReactRealworldExampleApp = require("../App.bs.js");

describe("App component", (function () {
        return Jest.test("renders without crashing", (function () {
                      var wrapper = Enzyme.mount(ReasonReact.element(undefined, undefined, App$ReasonReactRealworldExampleApp.make(/* array */[])));
                      return Jest.ExpectJs[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](wrapper));
                    }));
      }));

describe("makeLinkClass", (function () {
        Jest.test("append \"active\" if current and target is same ", (function () {
                var actual = App$ReasonReactRealworldExampleApp.makeLinkClass(/* Home */0, /* Home */0);
                return Jest.Expect[/* toBe */2]("nav-link active", Jest.Expect[/* expect */0](actual));
              }));
        return Jest.test("append empty string if current and target is not same", (function () {
                      var actual = App$ReasonReactRealworldExampleApp.makeLinkClass(/* Home */0, /* Login */1);
                      return Jest.Expect[/* toBe */2]("nav-link", Jest.Expect[/* expect */0](actual));
                    }));
      }));

/*  Not a pure module */
