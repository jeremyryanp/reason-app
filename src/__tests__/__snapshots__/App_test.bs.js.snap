// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without crashing 1`] = `
<div>
  <nav
    className="navbar navbar-light"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/#/"
      >
        conduit
      </a>
      <ul
        className="nav navbar-nav pull-xs-right"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link active"
            href="/#/"
          >
            Home
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="home-page"
  >
    <div
      className="banner"
    >
      <div
        className="container"
      >
        <h1
          className="logo-font"
        >
          conduit
        </h1>
        <p>
          A place to share your knowledge.
        </p>
      </div>
    </div>
    <div
      className="container page"
    >
      <div
        className="row"
      >
        <div
          className="col-md-9"
        >
          <div
            className="feed-toggle"
          >
            <ul
              className="nav nav-pills outline-active"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link active"
                  href="#"
                  onClick={[Function]}
                >
                  Global Feed
                </a>
              </li>
            </ul>
          </div>
          <div
            className="article-preview"
          >
            Initializing...
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="sidebar"
          >
            <p>
              Popular Tags
            </p>
            <div
              className="tag-list"
            >
              Initializing...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer>
    <div
      className="container"
    >
      <a
        className="logo-font"
        href="/"
      >
        conduit
      </a>
      <span
        className="attribution"
      >
        An interactive learning project from 
        <a
          href="https://thinkster.io"
        >
          Thinkster
        </a>
        . Code & design licensed under MIT.
      </span>
    </div>
  </footer>
</div>
`;
